### DISCUSSION TITLE
Question of circle window drawing algorithm on weak devices

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#235] [General]
At the pull-request #233, @ds-sloth have [noted](https://github.com/Wohlstand/TheXTech/pull/233#pullrequestreview-841641830) that there is a possible issue at circle window drawing on low-resource clients. However, I want to research this question deeper. I bet that shouldn't be hard if using OpenGL or similar APIs as this circle window can be drawn as a big polygon and no need to draw the thing pixel-by-pixel which is really heavy to process.

My algorithm with SDL Render API makes a trick by drawing a set of scan-lines:
https://github.com/Wohlstand/TheXTech/blob/6c5859c6b640d9c216e13ae0bde8af132f1db428/src/frm_main.cpp#L1763-L1817 that fills the space around the circle window. And later, separately, I draw 4 rectangles around this window.

**Comment by ds-sloth** [comment]
This scanlines method is interesting, on the 3DS that would mean ~240 GPU calls. I don't know if that is too expensive but we should test it. If it is too expensive, we could probably replace it with a low-vertex poly. The reason I was originally scared was that drawing *circles* (not poly approximations of circles) is very expensive on 3DS.

**Reply by Wohlstand** [reply]
The number of cycles depends on radius of the circle
